jax/_src/pretty_printer.py

```diff
@@ -69,12 +69,15 @@ def _can_use_color() -> bool:
 class Doc(util.StrictABC):
   __slots__ = ()
 
-  def format(self, width: int = 80, use_color: bool | None = None,
-             annotation_prefix=" # ") -> str:
+  def format(
+    self, width: int = 80, *, use_color: bool | None = None,
+    annotation_prefix: str = " # ",
+    source_map: list[list[tuple[int, int, Any]]] | None = None
```

Comment:
Consider explaining the format of `source_map`.

Reply:
Done.